src/clj/harja/kyselyt/sillat.sql

```diff
@@ -16,19 +16,15 @@ SET tyyppi          = :tyyppi,
     tr_numero       = :numero,
     tr_alkuosa      = :aosa,
     tr_alkuetaisyys = :aet,
-    siltaid         = :siltaid,
-    loppupvm        = :loppupvm,
-    lakkautuspvm    = :lakkautuspvm,
-    muutospvm       = :muutospvm,
-    status          = :status,
+    muutospvm       = :muutospvm::DATE,
+    loppupvm        = :loppupvm::DATE,
     poistettu       = :poistettu,
     kunnan_vastuulla = :kunnan-vastuulla,
     trex_oid        = :trex-oid,
     urakat          = (SELECT array_agg(a) ::INT[] || '{}' ::INT[] FROM unnest(ARRAY[:urakat] ::INT[]) a WHERE a IS NOT NULL),
     muokattu        = CURRENT_TIMESTAMP,
     muokkaaja       = (select id from kayttaja where kayttajanimi = 'Integraatio')
 WHERE (:trex-oid ::TEXT IS NOT NULL AND trex_oid = :trex-oid) OR
-      (trex_oid IS NULL AND :siltaid ::INT IS NOT NULL AND siltaid = :siltaid) OR
       (trex_oid IS NULL AND siltaid IS NULL AND (:tunnus ::TEXT IS NOT NULL AND siltatunnus = :tunnus) AND (:siltanimi ::TEXT IS NOT NULL AND siltanimi = :siltanimi));
```

Comment:
Pitäisköhän tämä trex_oid is null-mahdollisuus jättää pois? Se on ollu siirtymävaiheen ratkaisu ja siirtymästä on jo vuosia aikaa. Aineistossa ei varmaan oo siltoja, joilla trex_oidia ei oo, koska se on niiden nykyinen yksilöivä tunnus, joka korvasi tuon siltaid:n.

Reply:
Hyvä idea, mutta jos tämän jättää, sen pitäisi korjata muutama silta, joka näkyy tällä hetkellä harjassa väärin.
Kannassa on 11 siltaa ilman trex_oidia, jotka eivät ole poistettu, päättyneet tai lakkautettu.
Ainakin kaksi ensimmäistä on sellaisia, jotka löytyy samalla sijainnilla ja nimellä voimassaolevana tuoreemmin päivitettynä. Aineistossa nämä sillat ilman trex_oidia näkyvät poistettuna.

Jos tämän jättää, nuo sillat merkittäisiin poistetuiksi kantaan, mutta tokihan ne voisi merkitä poistetuksi myös käsin. Kumpaa mieltä itse olet?

Reply:
Tee miten susta tuntuu varmemmalta.